upath/core.py

```diff
@@ -155,9 +128,10 @@ def __new__(cls, *args, **kwargs):
                 args = tuple(args_list)
                 self = cls._from_parts_init(args, init=False)
                 self._init(*args, **kwargs)
-        else:
-            self = super().__new__(*args, **kwargs)
-        return self
+                return self
+
+        # treat as local filesystem, return PosixPath or WindowsPath
+        return pathlib.Path(*args, **kwargs)
```

Comment:
This is slightly problematic, as I normally would expect this to be an instance of UPath (or a subclass), but it isn't. However, this seems to be the option with the easiest usability. It would be great if we would add the correct type-annotation `-> Union[UPath, Path]` at some point, since this is honored by mypy (but that's a different issue). The related discussion in mypy is quite interesting, see https://github.com/python/mypy/issues/1020.

Reply:
I agree that this may be slightly odd. However, it is very convenient. Also, this is actually the current behavior of `UPath` just implemented differently.